src/main/java/com/example/java/test/junior/developer/service/UserService.java

```diff
@@ -0,0 +1,60 @@
+package com.example.java.test.junior.developer.service;
+
+import com.example.java.test.junior.developer.dto.UserDto;
+import com.example.java.test.junior.developer.dto.UserRequestDto;
+import com.example.java.test.junior.developer.mapper.UserMapper;
+import com.example.java.test.junior.developer.model.User;
+import com.example.java.test.junior.developer.repository.UserRepository;
+import com.example.java.test.junior.developer.security.KeycloakAdminClient;
+import lombok.RequiredArgsConstructor;
+import org.keycloak.adapters.springboot.KeycloakSpringBootProperties;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserMapper userMapper;
+    private final UserRepository userRepository;
+    private final KeycloakAdminClient keycloakAdminClient;
+    private KeycloakSpringBootProperties keycloakProperties;
+
+    @Transactional
+    public UserDto createUser(UserRequestDto userRequestDto) {
+        var user = userMapper.toUserRepresentation(userRequestDto);
+        keycloakAdminClient.createUser(keycloakProperties.getRealm(), user);// Не можу зрозуміти як "правильно" передати realm при створенні юзера.
```

Comment:
You should not pass realm from here. You should have it application.properties (and mapped in ConfigurationProperties class)